qa/L0_lifecycle/lifecycle_test.py

```diff
@@ -826,5 +826,196 @@ def test_dynamic_file_delete(self):
                 except InferenceServerException as ex:
                     self.assertTrue(False, "unexpected error {}".format(ex))
 
+    def test_model_control(self):
+        input_size = 16
+        models_base = ('graphdef',)
+        models_shape = ((input_size,),)
+        models = list()
+        ensemble_prefix = "simple_"
+        ensemble_models = list()
+        for m in models_base:
+            model_name = tu.get_model_name(m, np.float32, np.float32, np.float32)
+            models.append(model_name)
+            ensemble_models.append(ensemble_prefix + model_name)
+
+        # Make sure the models are not in the status
+        for model_name in (models + ensemble_models):
+            for pair in [("localhost:8000", ProtocolType.HTTP), ("localhost:8001", ProtocolType.GRPC)]:
+                try:
+                    ctx = ServerStatusContext(pair[0], pair[1], model_name, True)
+                    ss = ctx.get_server_status()
+                    self.assertTrue(False, "expected unknown model failure")
+                except InferenceServerException as ex:
+                    self.assertEqual("inference:0", ex.server_id())
+                    self.assertGreater(ex.request_id(), 0)
+                    self.assertTrue(
+                        ex.message().startswith("no status available for unknown model"))
+
+        # Load non-existing model
+        for pair in [("localhost:8000", ProtocolType.HTTP), ("localhost:8001", ProtocolType.GRPC)]:
+            try:
+                ctx = ModelControlContext(pair[0], pair[1], True)
+                ctx.load("unknown_model")
+                self.assertTrue(False, "expected unknown model failure")
+            except InferenceServerException as ex:
+                self.assertEqual("inference:0", ex.server_id())
+                self.assertGreater(ex.request_id(), 0)
+                self.assertTrue(
+                    ex.message().startswith("failed to stat directory for model"))
+
+        # Load ensemble model first, should fail due to depending model is not loaded
```

Comment:
If they load an ensemble model we should automatically  load all the dependent models. Otherwise user has to discover all the necessary models and figure out the right order to load them.

Reply:
Sounds good. Should I make the change on a separate PR?

Reply:
Yes, that will be ok.